src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardInputActionObserver.cs

```diff
@@ -200,6 +200,99 @@ public class KeyboardInputActionObserver : MonoBehaviour
             { "OEM4", Key.OEM4 },
             { "OEM5", Key.OEM5 },
         });
+        
+        // Key -> (without-Shift, with-Shift) values
+        public static readonly IReadOnlyDictionary<Key, (char, char)> KeyToValueMap = new ReadOnlyDictionary<Key, (char,char)>(new Dictionary<Key, (char,char)>()
+        {
+            //row 1 (top row)
```

Comment:
For "standard" keyboards this is the function row. Need to do some testing to see whether we need to handle fn keys.

Reply:
Exactly. We need to select the correct character based on whether `shift` is being held at the same time. I'm sure we could clean up the dictionaries at some point, but we wanted to get this working for our partner project ASAP.